tracer/src/Datadog.Trace/Logging/DirectSubmission/Formatting/LogFormatter.cs

```diff
@@ -26,28 +28,54 @@ internal class LogFormatter
         private readonly string? _source;
         private readonly string? _service;
         private readonly string? _host;
-        private readonly string? _globalTags;
         private readonly string? _env;
         private readonly string? _version;
+        private readonly IGitMetadataTagsProvider _gitMetadataTagsProvider;
+        private string? _tags;
+        private bool _gitMetadataAdded;
 
         private string? _ciVisibilityDdTags;
 
         public LogFormatter(
             ImmutableDirectLogSubmissionSettings settings,
             string serviceName,
             string env,
-            string version)
+            string version,
+            IGitMetadataTagsProvider gitMetadataTagsProvider)
         {
+            _gitMetadataTagsProvider = gitMetadataTagsProvider;
             _source = string.IsNullOrEmpty(settings.Source) ? null : settings.Source;
             _service = string.IsNullOrEmpty(serviceName) ? null : serviceName;
             _host = string.IsNullOrEmpty(settings.Host) ? null : settings.Host;
-            _globalTags = string.IsNullOrEmpty(settings.GlobalTags) ? null : settings.GlobalTags;
+            _tags = string.IsNullOrEmpty(settings.GlobalTags) ? null : settings.GlobalTags;
             _env = string.IsNullOrEmpty(env) ? null : env;
             _version = string.IsNullOrEmpty(version) ? null : version;
         }
 
         internal delegate LogPropertyRenderingDetails FormatDelegate<T>(JsonTextWriter writer, in T state);
 
+        private void EnrichTagsStringWithGitMetadata()
+        {
+            if (_gitMetadataAdded)
+            {
+                return;
+            }
+
+            if (!_gitMetadataTagsProvider.TryExtractGitMetadata(out var gitMetadata))
+            {
+                // no git tags found, we can try again later
+                return;
+            }
+
+            if (gitMetadata != GitMetadata.Empty)
+            {
+                var gitMetadataTags = $"{CommonTags.GitCommit}:{gitMetadata.CommitSha},{CommonTags.GitRepository}:{gitMetadata.RepositoryUrl}";
```

Comment:
Could there be escaping issues related to `:` in the URL here? 🤔

Reply:
There might :/ I went ahead and removed the scheme entirely in 058285aa563a30ab74170ac3559cf8162a139c73 .  

I think there's no reason not to remove the scheme entirely here because the Source Code Integration backend supports that; in fact, our [in-app onboarding page](https://app.datadoghq.com/source-code/setup/) actively tries to force customers to omit it when specifying the repository url through `DD_TAGS`, because historically, many tracers had nasty escaping-related bugs where not omitting caused things to fail.